.github/workflows/on_PR_meson.yaml

```diff
@@ -126,6 +127,36 @@ jobs:
           meson setup "${{github.workspace}}/build" -Dauto_features=${{matrix.deps}} -Dwarning_level=3 -Dcpp_std=c++20
           meson compile -C "${{github.workspace}}/build" --verbose
           meson test -C "${{github.workspace}}/build" --verbose
+  Cygwin:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        deps: ['enabled', 'disabled']
+        platform: ['x86', 'x86_64']
+    name: Cygwin-${{matrix.platform}}-deps=${{matrix.deps}}
+    defaults:
+      run:
+        shell: C:\cygwin\bin\bash.exe --noprofile --norc -o igncr -eo pipefail '{0}'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: cygwin/cygwin-install-action@v4
+        with:
+          platform: ${{matrix.platform}}
+          packages: :>
+            ca-certificates
+            cmake
+            gcc-g++
```

Comment:
Try [`gcc-g++=13.2.0+20230902-0.1`](https://www.cygwin.com/packages/summary/gcc-g++.html)

Reply:
```
meson.build:1:0: ERROR: Unknown compiler(s): [['g++-13']]
The following exception(s) were encountered:
Running `g++-13 --version` gave "[Errno 2] No such file or directory: 'g++-13'"
```

This is making me want to give up and just use MSYS2 instead.

Reply:
I'm impressed by your persistence so far 😉

Let's just comment out Cygwin and bring it back when gcc13 becomes current there.

Reply:
Not giving up yet :)

I tested compiling fmt on Cygwin. Didn't work. I'll use MSYS2 instead, which is Cygwin without old compilers/packages.

Reply:
Finally got this damn thing building. Some of the wraps are broken. I'll reintroduce that after I fix.

Reply:
Re Cygwin: still no insights when they might move from GCC 11, and [packaging fmt seem not to be a [](priority](https://cygwin.com/pipermail/cygwin/2023-November/254865.html).)